src/cogs/rules.py

```diff
@@ -81,70 +81,73 @@ async def on_member_join(self, member):
-                while len(wizard_msg) > self.file["max_msg_size"]:
-                    chunk = wizard_msg[0:self.file["max_msg_size"]]
-                    end_index = chunk.rfind('\n')
-                    messages.append(chunk[0:end_index])
-                    wizard_msg = wizard_msg[end_index:]
+                    messages = []
+                    while len(wizard_msg) > self.file["max_msg_size"]:
+                        chunk = wizard_msg[0:self.file["max_msg_size"]]
+                        end_index = chunk.rfind('\n')
+                        messages.append(chunk[0:end_index])
+                        wizard_msg = wizard_msg[end_index:]
 
-                messages.append(wizard_msg)
+                    messages.append(wizard_msg)
 
-                reply = msg
-                for message in messages:
-                    reply = await reply.reply(message)
+                    reply = msg
+                    for message in messages:
+                        reply = await reply.reply(message)
 
-                await reply.add_reaction(self.file["no_react"])
-                await reply.add_reaction(self.file["yes_react"])
+                    await reply.add_reaction(self.file["no_react"])
+                    await reply.add_reaction(self.file["yes_react"])
 
-                reaction, user = await self.bot.wait_for('reaction_add',
+                    reaction, user = await self.bot.wait_for('reaction_add',
                         check=lambda reaction, user: _check(reaction, user, reply),
                         timeout=self.file["massjoin_wizard_timeout"])
 
-                if reaction.emoji == self.file["no_react"]:
-                    await reply.reply("Not banning any users and resetting the join detection")
+                    if reaction.emoji == self.file["no_react"]:
+                        await reply.reply("Not banning any users and resetting the join detection")
 
-                if reaction.emoji == self.file["yes_react"]:
-                    for user in self._recent_joins:
-                        if user["assumed_bot"]:
-                            await member.guild.ban(
+                    if reaction.emoji == self.file["yes_react"]:
+                        for user in self._recent_joins:
+                            if user["assumed_bot"]:
+                                await member.guild.ban(
                                     self.bot.get_user(user["id"]) if self.bot.get_user(user["id"])
                                         else await self.bot.fetch_user(user["id"]))
 
-                    await reply.reply("Banned the bots")
+                        await reply.reply("Banned the bots")
 
-                await reply.clear_reactions()
+                    await reply.clear_reactions()
 
-            await msg.clear_reactions()
+                await msg.clear_reactions()
 
-            self._recent_joins.clear()
-            self.massjoin_active = False
+            finally:
```

Comment:
Should probably catch it and send the exception if it occurs so we know whether to manually ban or not?

Reply:
This was mostly intended for just making sure the reset code is run.
With the main reason being the actually relatively likely to fail `wait_for()`s and everything else going properly being an assumption.

So there's no logic for ensuring that anything special going wrong being handled as
- when the actual banning fails, the success message doesn't appear
- when the notification message isn't sent, the admins will notice anyway
- when the ban "wizard" message[s] aren't sent, the admins will notice that too and can take action manually

Reply:
I'd like visual feedback if the banning fails, the rest is fine